lib/TH/THHalf.c

```diff
@@ -76,37 +91,31 @@ float TH_half2float(THHalf h)
         exponent += 0x70;
     }
 
-    int temp = ((sign << 31) | (exponent << 23) | mantissa);
-    float x;
-    memcpy(&x,&temp,sizeof(float));
-    return x;
+    *(unsigned*)res = ((sign << 31) | (exponent << 23) | mantissa);
 }
 
-THHalf TH_float2half(float f)
+void TH_float2halfbits(float* src, unsigned short* dest)
 {
-    THHalf ret;
-
-    unsigned x;
-    memcpy(&x,&f,sizeof(f));
-    unsigned u = (x & 0x7fffffff), remainder, shift, lsb, lsb_s1, lsb_m1;
+    unsigned u = *(unsigned*)src;
```

Comment:
you are missing  & 0x7fffffff, see the old version.  This causes test/test_half.lua to fail.